docs/content/guides/developer/app-examples/escrow.mdx

```diff
@@ -3,4 +3,160 @@ title: Escrow
 draft: true
 ---
 
-Content coming soon
\ No newline at end of file
+This guide demonstrates an atomic swap, which does not require a trusted third party, by utilizing a [shared object](/concepts/object-ownership/shared) as an escrow. Shared objects are a unique concept to Sui. They are able to be modified by any transaction and by any signer, given they meet the requirements set forth by the transaction. For more information on shared objects versus owned objects, see [here](/guides/developer/sui-101/shared-owned).
+
+In addition to the smart contract, this example also includes:
+
+- A frontend for people to list object for sale and to accept trades
+- A service that indexes chain state in order to discover trades
+
+## Prerequisites
+
+Before getting started, make sure you have:
+
+- [Installed the latest version of Sui](/guides/developer/getting-started/sui-install).
+- [Configured a valid network environment](/references/cli/client#set-current-environment), as we will be deploying the module on testnet.
+
+## Backend
+
+This particular protocol consists of three phases:
+
+1. One party `lock`s their object, obtaining a `Locked` object and its `Key`. This party can `unlock` their object to preserve liveness if the other party stalls before completing the second stage.
+2. The other party registers a publicly accessible, shared `Escrow` object. This effectively locks their object at a particular version as well, waiting for the first party to complete the swap. The second party is able to request their object is returned to them, to preserve liveness as well.
+3. The first party sends their locked object and its key to the shared `Escrow` object. This completes the swap, as long as all conditions are met: The sender of the swap transaction is the recipient of the `Escrow`, the key of the desired object (`exchange_key`) in the escrow matches the key supplied in the swap, and the key supplied in the swap unlocks the `Locked<U>`.
+
+Let’s create a Sui project using the terminal command `sui move new escrow`. Create a file in the sources directory named `shared.move`, and let’s go through the code together.
+
+See [this](https://github.com/MystenLabs/sui/blob/main/examples/move/escrow/sources/shared.move) for the complete code.
+
+### `shared.move`
+
+Let's go through the code line by line:
+
+#### Structs
+
+##### `Escrow<T>`
+
+This struct represents an object held in escrow. It contains the following fields:
+
+- `id`: Unique identifier for the escrow object.
+- `sender`: Address of the owner of the `escrowed` object.
+- `recipient`: Intended recipient of the `escrowed` object.
+- `exchange_key`: ID of the key that opens the lock on the object sender wants from the recipient.
```

Comment:
Worth explaining here that we use the ID of the `key` rather than the ID of the object because we want to make sure that the object is not modified after we initiate a trade.